### ISSUE TITLE
Documentation revamp

### ISSUE THREAD

**Issue by koddsson** [chaijs/chai#1595]
I really like the documentation and I want to really try to avoid rewriting it's system but I think we should colocate it with the code in this repo as opposed to keeping it in a different repo. I also think we should generate most of the API documentation from the JSDocs. This way we don't change the code without updating the docs as well.

Here's a loose list of what I think we should do:

- Install ESLint and the `jsdoc` eslint plugin to validate that the jsdoc comments are in the correct format.
- Fix all the ESLint `jsdoc` issues.
- Use some sort of generator to generate the API site. I'd be happy to use Jekyll but maybe something like 11ty would suit us better?
- Add a CI check that validates that documentation is correct somehow or even deploys to a temporary host.
- Migrate the docs from the old repo to this repo and the new docs system.
- ???

What do you think?

@chaijs/core

**Comment by koddsson** [comment]
Here's a quick thing I threw up using jsdoc:

![image](https://github.com/chaijs/chai/assets/318208/adeea3b8-7c78-4031-b78c-fb8ce8d0f274)

We could obviously tweak the styles and all that. And also make the markdown work and update all the docs like you said.

**Comment by koddsson** [comment]
I can fiddle with this some more and maybe put on a draft PR.

**Comment by koddsson** [comment]
@keithamus @43081j; I started writing an eleventy site derived from the JSDoc info and the chai plugins exposed in the npm registry API. I'm using [PicoCSS](https://picocss.com/) for styling.

What do you think? I wanna ask now so I don't waste my time if you feel this isn't worth it. 

Home page:
![Image](https://github.com/user-attachments/assets/729b21fd-71bf-4320-aa18-496b5b6365ad)

API method example:
![Image](https://github.com/user-attachments/assets/fca650a4-303e-45f6-a17e-b490f341a3a2)

I've also got fetching the plugins and outputting pages for them up and running, I just haven't started rendering out those yet.

**Comment by keithamus** [comment]
YES! Let's do it!

**Comment by 43081j** [comment]
looks great. would love to sort out our docs for sure

they've gone a little stale over time. would be great to do this

**Comment by koddsson** [comment]
I've gotten most things working, _but_ I was hoping to generate API docs from the JSDoc comments (and then eventually from TypeScript types as well). But all the examples are just for `expect` (see [`fail`](https://github.com/chaijs/chai/blob/93409cd3f7975f67cae95c780c81a056eed364ac/lib/chai/interface/assert.js#L31-L51) for example).

I want to derive the docs from the code so that we don't have as much drift from the docs and the actual code. But I might just punt on that for now and crib the old API docs, then I can start working on reworking the JSDoc comments to be more "parseable", with examples for `should` and `assert` and tests that assert that any given assertion has all the required examples and metadata.

**Comment by koddsson** [comment]
[Here's the repo](https://github.com/koddsson/new-chai-docs) for what I got so far and a built version hosted on https://koddsson.github.io/new-chai-docs. I think it's pretty nifty, but happy to hear any and all feedback :)

**Comment by 43081j** [comment]
is there a way we can have some kind of contents/navigation in the API docs?

for example, here:
https://koddsson.github.io/new-chai-docs/api/bdd/

it'd be good to have the full list of headings and be able to jump to them

**Comment by koddsson** [comment]
> is there a way we can have some kind of contents/navigation in the API docs?
> 
> for example, here: https://koddsson.github.io/new-chai-docs/api/bdd/
> 
> it'd be good to have the full list of headings and be able to jump to them

Yup! I have that working for https://koddsson.github.io/new-chai-docs/guide/ and you can use the navigation at the top of the page to jump to whatever you'd like. Should be trivial to implment for the API page as well.

**Comment by koddsson** [comment]
@43081j @keithamus; What do you think about this? I can see if I can migrate the website over if there are no show-stoppers.

**Comment by keithamus** [comment]
I think I'd like to see some of the branding carried over. I am not saying it needs to look identical but it might be disruptive/surprising for users to visit the new site which looks so drastically different. Thoughts?

**Comment by koddsson** [comment]
> I think I'd like to see some of the branding carried over. I am not saying it needs to look identical but it might be disruptive/surprising for users to visit the new site which looks so drastically different. Thoughts?

I can work on that! 

Are you thinking more about assets, theming or layout? Some combination of the three? Or something completely different?

**Comment by keithamus** [comment]
Assets & theming for sure. For example incorporating the Chai logo, perhaps the backgrounds, a similar colour theme.  Modernising the website is fine by me but I think it should carry over the identity.
